File: Entities/Notification.cs
namespace PrivateHospitalSystem.Entities
{
    public class Notification
    {
        public Guid Id { get; set; }

        public Guid? PatientId { get; set; }
        public Patient? Patient { get; set; }

        // Allow notifications to target a doctor as well
        public Guid? DoctorId { get; set; }
        public Doctor? Doctor { get; set; }

        public string Message { get; set; } = string.Empty;
        public bool IsRead { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/src/types/notification.ts
export interface NotificationItem {
  id: string
  userId?: string
  patientId?: string
  doctorId?: string
  title: string
  message: string
  type: string
  isRead: boolean
  createdAt: string
}

export interface CreateNotificationDto {
  patientId?: string
  doctorId?: string
  title: string
  message: string
  type: string
}
